src/plugins/github/fetchGithubRepo.js

```diff
@@ -76,6 +79,23 @@ export default async function fetchGithubRepo(
   return ((mirror.extract(resolvedId): any): Repository);
 }
 
+// GitHub object IDs are urlsafe-base64-encoded strings that decode to
+// ASCII strings of the form "123:Typename4567[...]", where the "123"
+// numbers are a function only of the typename and the "4567" numbers
+// are the object's database ID, and the "[...]" is either empty or a
+// further section like ":commithash" for commits.
+//
+// See tests for `_guessTypename` for some example object IDs.
+const GITHUB_ID_TYPENAME_PATTERN = /^[0-9]*:([A-Za-z0-9_-]*[A-Za-z_-])[0-9]+(?:[^A-Za-z0-9_-].*)?$/;
+
+export function _guessTypename(
+  objectId: Schema.ObjectId
+): Schema.Typename | null {
+  const decodedId = new Buffer(objectId, "base64").toString("utf-8");
+  const match = decodedId.match(GITHUB_ID_TYPENAME_PATTERN);
```

Comment:
In theory the above binary to utf8 conversion would allow access to the full unicode range, including control characters. Do we know if [`String.prototype.match()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match) is vulnerable to any [naughty strings](https://github.com/minimaxir/big-list-of-naughty-strings)?

Particularly during execution, as the regex used looks restrictive enough to avoid naughty strings in the output.

Reply:
It’d be quite surprising if the builtin `match` had inherent structural
vulnerabilities, as opposed to something like ReDoS (which this regex is
not vulnerable to). Even if there were a particular naughty string that
caused this regex to behave unexpectedly, it seems clear that the worst
that could happen would be that this function returns the wrong result.
Recall that we JSON-stringify guessed typenames before printing them to
the console, which neutralizes control characters.